tap_kafka/sync.py

```diff
@@ -182,20 +185,26 @@ def kafka_message_to_singer_record(message, primary_keys):
         "message_timestamp": get_timestamp_from_timestamp_tuple(message.timestamp()),
     }
 
-    # Add primary keys to the record message
-    for key in primary_keys:
-        pk_selector = primary_keys[key]
-        try:
-            record[key] = dpath.util.get(message.value(), pk_selector)
-        except KeyError:
-            raise PrimaryKeyNotFoundException(f"Custom primary key not found in the message: '{pk_selector}'")
+    # Add PKs to the record. In case custom PKs are defined, use them
+    if primary_keys:
+        for key in primary_keys:
+            pk_selector = primary_keys[key]
+            try:
+                record[key] = dpath.util.get(message.value(), pk_selector)
+            except KeyError:
+                raise PrimaryKeyNotFoundException(f"Custom primary key not found in the message: '{pk_selector}'")
+    elif use_message_key:
+        if not message.key():
+            raise PrimaryKeyNotFoundException("Kafka message key not found in the message")
+        # message.key() can return string or bytes, so extra check to accommodate with either
+        record['message_key'] = message.key() if isinstance(message.key(), str) else message.key().decode('utf-8')
```

Comment:
> message.key().decode('utf-8')

is the message key always utf8 encoded?

Reply:
overall, not necessarily, but as a somewhat common practice - yes. If it was intended to be a string, then most certainly.

It is _theoretically_ possible to handle other variations as well, e.g. when key is also using protobuf format, however that would require other potentially substantial code changes to either introduce option to provide key deserialisation similarly to how protobuf value deserd one is handled or by integrating schema registry.

Previously discussed w/ @koszti and agreed that it would be certainly an improvement in long term, but as of now it would make sense to provide explicit restriction in docs and throw unhandled `UnicodeDecodeError`.